### PR TITLE
fix: creating hyperv machine in airgap now does not try to use wsl image

### LINKED ISSUE #9314: Cannot create Windows Hyperv Podman Machine with airgapped Podman Desktop with access to internet
### Bug description

Creating podman machine with hyperv provider with airgap podman desktop with access to internet throws an error.

Image

### Operating system

Windows 10

### Installation Method

Other

### Version

1.13.0

### Steps to reproduce

0. Have HyperV enabled, no network restriction
1. Install/start podman desktop airgapped (1.13.2)
2. Create podman machine with hyperv
3. Assert: Machine is created
Actual result: Error creating the image

### Relevant log output

```
Flag --image-path has been deprecated, use --image instead

Error: failed to add resource: Job failed with error code: 32768 ('podman-machine-default' failed to add device 'Virtual Hard Disk'. (Virtual machine ID 61AF76EC-1B4F-4FFB-AFC7-7004E0283682)  Failed to open attachment 'C:\Users\Ondrej\.local\share\containers\podman\machine\hyperv\podman-machine-default-amd64.vhdx'. Error: 'The file or directory is corrupted and unreadable.'.  'podman-machine-default': Cannot get information for attachment 'C:\Users\Ondrej\.local\share\containers\podman\machine\hyperv\podman-machine-default-amd64.vhdx'. (Virtual machine ID 61AF76EC-1B4F-4FFB-AFC7-7004E0283682)  Failed to open attachment 'C:\Users\Ondrej\.local\share\containers\podman\machine\hyperv\podman-machine-default-amd64.vhdx'. Error: 'The file or directory is corrupted and unreadable.'.)
```

### Additional context

After we have fixed https://github.com/containers/podman-desktop/issues/9296. We have now regression in this issue, where we cannot create hyperv at all with airgapped installer.

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Adds support for airgapped hyperv
Relevant PR: https://github.com/containers/podman-desktop/pull/9303 (will be rebased based on this changes)

- [x] rebase after https://github.com/containers/podman-desktop/pull/9708 will be merged 

### What issues does this PR fix or reference?
Closes #9314 

### How to test this PR?
Try to create a HyperV machine in airgap mode:
1. pnpm install
2. AIRGAP_DOWNLOAD=true pnpm compile:current --win nsis
3. run created airgap instalator with admin
4. create HyperV machine - should end with error dial TCP... 125 error code (~ does not have an access to internet)
- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by gastoner** [comment]
TODO: Rebase after https://github.com/containers/podman-desktop/pull/9708 will be merged

**Comment by jeffmaury** [comment]
I have an error when trying to create an Hyper-V machine:

image

**Comment by odockal** [comment]
@gastoner Please rebase.

**Comment by jeffmaury** [comment]
LGTM. Is there a way we can add a unit test for the added use case ?

**Comment by gastoner** [comment]
@jeffmaury added

File: extensions/podman/packages/extension/src/extension.spec.ts [hunk 1]
```diff
@@ -646,6 +646,45 @@ describe.each(['macos', 'windows'])('verify create on %s', os => {
   });
 });
 
+test('verify create command called in airgap mode will try to download hyperv image', async () => {
+  vi.mocked(extensionApi.env).isWindows = true;
```
Comment (jeffmaury) [inline extensions/podman/packages/extension/src/extension.spec.ts]: I don't get why isWindows is handled differently that isMac I think these are functions

**Review by odockal — APPROVED** [review]
On windows 10 + Hyperv it works well. When airgapped Pd is used, I can go offline and I can still install podman. If I try to create a Hyperv Machine, it complains about connection. I can create hyperv podman machine once I back online. 
WSL Machine can be created when in offline.

**Comment by gastoner** [comment]
@jeffmaury I've updated the test, but there are some ifs in the test due to fact that windows has WSL and HyperV

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**